src/main/java/it/smartcommunitylab/aac/webauthn/provider/WebAuthnAttributeProvider.java

```diff
@@ -0,0 +1,117 @@
+package it.smartcommunitylab.aac.webauthn.provider;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import org.springframework.util.Assert;
+
+import it.smartcommunitylab.aac.SystemKeys;
+import it.smartcommunitylab.aac.attributes.BasicAttributesSet;
+import it.smartcommunitylab.aac.common.NoSuchUserException;
+import it.smartcommunitylab.aac.core.auth.UserAuthenticatedPrincipal;
+import it.smartcommunitylab.aac.core.base.AbstractProvider;
+import it.smartcommunitylab.aac.core.base.ConfigurableProperties;
+import it.smartcommunitylab.aac.core.base.DefaultUserAttributesImpl;
+import it.smartcommunitylab.aac.core.model.UserAttributes;
+import it.smartcommunitylab.aac.core.provider.AttributeProvider;
+import it.smartcommunitylab.aac.webauthn.model.WebAuthnUserAuthenticatedPrincipal;
+import it.smartcommunitylab.aac.webauthn.persistence.WebAuthnUserAccount;
+import it.smartcommunitylab.aac.webauthn.service.WebAuthnUserAccountService;
+
+public class WebAuthnAttributeProvider extends AbstractProvider implements AttributeProvider {
+    private final WebAuthnUserAccountService userAccountService;
+    private final WebAuthnIdentityProviderConfig providerConfig;
+
+    public WebAuthnAttributeProvider(
+            String providerId,
+            WebAuthnUserAccountService userAccountService,
+            WebAuthnIdentityProviderConfig providerConfig,
+            String realm) {
+        super(SystemKeys.AUTHORITY_WEBAUTHN, providerId, realm);
+        Assert.notNull(userAccountService, "account service is mandatory");
+        Assert.notNull(providerConfig, "provider config is mandatory");
+
+        this.userAccountService = userAccountService;
+        this.providerConfig = providerConfig;
+    }
+
+    @Override
+    public String getType() {
+        return SystemKeys.RESOURCE_ATTRIBUTES;
+    }
+
+    @Override
+    public String getName() {
+        return providerConfig.getName();
+    }
+
+    @Override
+    public String getDescription() {
+        return providerConfig.getDescription();
+    }
+
+    @Override
+    public ConfigurableProperties getConfiguration() {
+        return providerConfig;
+    }
+
+    @Override
+    public Collection<UserAttributes> getAttributes(String subjectId) {
+        // we expect subjectId to be == userId
+        String userId = subjectId;
+        String username = parseResourceId(userId);
+        String provider = getProvider();
+
+        WebAuthnUserAccount account;
+        try {
```

Comment:
remove the try/catch, a "find" method should return null on not found, no exception to catch

Reply:
Fixed in b8f7787656af7aa75e456a147cf9f188dbd1b70a